### ISSUE TITLE
tickFormat and custom locale

### ISSUE THREAD

**Issue by rahulpowar** [d3/d3-scale#70]
Currently, https://github.com/d3/d3-scale/blob/master/src/tickFormat.js only supports the U.S. English locale.

Would it make sense to optionally pass this function a locale object?

**Comment by mbostock** [comment]
Perhaps. The function you linked is an internal function, so you’d need to expose the locale argument to the public methods such as *linear*.tickFormat, *log*.tickFormat, etc. 

Another option would be a new method to set the default locale on [d3-format](https://github.com/d3/d3-format), thereby changing the behavior of the default locale methods ([d3.format](https://github.com/d3/d3-format#format) and [d3.formatPrefix](https://github.com/d3/d3-format#formatPrefix)). You’d also want a similar method on [d3-time-format](https://github.com/d3/d3-time-format).

**Comment by rahulpowar** [comment]
I guess being able to set the default in `d3-format` and `d3-time-format` would normalize the behaviour across all usage which seems desirable.

**Comment by mbostock** [comment]
d3.formatDefaultLocale added in d3/d3-format@edb73d060994288716425a47e36b84ee7a294587.

**Comment by mbostock** [comment]
d3.timeFormatDefaultLocale added in d3/d3-time-format@5a913f0600f973fc8756a33f89700fe6db7a65aa. I will push new releases shortly.

**Comment by ljani** [comment]
While this works localizing the strings, [the formats are still in en-US](https://github.com/d3/d3-scale/blob/ab2c5d01f714668f97250c66c08a9e33182dc7bd/src/time.js#L30-L37). Any opinions how to localize this? Sure I can duplicate [the `tickFormat`](https://github.com/d3/d3-scale/blob/ab2c5d01f714668f97250c66c08a9e33182dc7bd/src/time.js#L60-L68) functionality, but it would be nice to be able to specify the formats directly.

**Comment by mbostock** [comment]
@ljani Yes, if you read the [d3-time-format README](https://github.com/d3/d3-time-format/blob/master/README.md#d3-time-format), you’ll see it recommends an implementation for a multi-scale time format.

**Comment by ljani** [comment]
Yeah, I saw it, so let me rephrase.. I mean I was hoping for a better way than looking up that piece of code and copy-pasting that code each time I create a new project. Something similar like requiring a json file and just calling `d3.timeScaleDefaultLocale(locale)`.

Thanks though!

**Comment by mbostock** [comment]
You could always release your own D3 plugin! 😜

https://bost.ocks.org/mike/d3-plugin/
